File: SampleCode/RestApi/Clients/Java/src/main/java/io/swagger/client/model/PluginsPluginInfo.java
/*
 * Emby Server REST API
 * 
 */

package io.swagger.client.model;

import java.util.Objects;
import java.util.Arrays;
import com.google.gson.TypeAdapter;
import com.google.gson.annotations.JsonAdapter;
import com.google.gson.annotations.SerializedName;
import com.google.gson.stream.JsonReader;
import com.google.gson.stream.JsonWriter;
import io.swagger.v3.oas.annotations.media.Schema;
import java.io.IOException;
/**
 * This is a serializable stub class that is used by the api to provide information about installed plugins.  
 */
@Schema(description = "This is a serializable stub class that is used by the api to provide information about installed plugins.  ")

public class PluginsPluginInfo {
  @SerializedName("Name")
  private String name = null;

  @SerializedName("Version")
  private String version = null;

  @SerializedName("ConfigurationFileName")
  private String configurationFileName = null;

  @SerializedName("Description")
  private String description = null;

  @SerializedName("Id")
  private String id = null;

  @SerializedName("ImageTag")
  private String imageTag = null;

  public PluginsPluginInfo name(String name) {
    this.name = name;
    return this;
  }

   /**
   * The name.
   * @return name
  **/
  @Schema(description = "The name.")
  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public PluginsPluginInfo version(String version) {
    this.version = version;
    return this;
  }

   /**
   * The version.
   * @return version
  **/
  @Schema(description = "The version.")
  public String getVersion() {
    return version;
  }

  public void setVersion(String version) {
    this.version = version;
  }

  public PluginsPluginInfo configurationFileName(String configurationFileName) {
    this.configurationFileName = configurationFileName;
    return this;
  }

   /**
   * The name of the configuration file.
   * @return configurationFileName
  **/
  @Schema(description = "The name of the configuration file.")
  public String getConfigurationFileName() {
    return configurationFileName;
  }

  public void setConfigurationFileName(String configurationFileName) {
    this.configurationFileName = configurationFileName;
  }

  public PluginsPluginInfo description(String description) {
    this.description = description;
    return this;
  }

   /**
   * The description.
   * @return description
  **/
  @Schema(description = "The description.")
  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }

  public PluginsPluginInfo id(String id) {
    this.id = id;
    return this;
  }

   /**
   * The unique id.
   * @return id
  **/
  @Schema(description = "The unique id.")
  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }

  public PluginsPluginInfo imageTag(String imageTag) {
    this.imageTag = imageTag;
    return this;
  }

   /**
   * The image URL.
   * @return imageTag
  **/
  @Schema(description = "The image URL.")
  public String getImageTag() {
    return imageTag;
  }

  public void setImageTag(String imageTag) {
    this.imageTag = imageTag;
  }


  @Override
  public boolean equals(java.lang.Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    PluginsPluginInfo pluginsPluginInfo = (PluginsPluginInfo) o;
    return Objects.equals(this.name, pluginsPluginInfo.name) &&
        Objects.equals(this.version, pluginsPluginInfo.version) &&
        Objects.equals(this.configurationFileName, pluginsPluginInfo.configurationFileName) &&
        Objects.equals(this.description, pluginsPluginInfo.description) &&
        Objects.equals(this.id, pluginsPluginInfo.id) &&
        Objects.equals(this.imageTag, pluginsPluginInfo.imageTag);
  }

  @Override
  public int hashCode() {
    return Objects.hash(name, version, configurationFileName, description, id, imageTag);
  }


  @Override
  public String toString() {
    StringBuilder sb = new StringBuilder();
    sb.append("class PluginsPluginInfo {\n");
    
    sb.append("    name: ").append(toIndentedString(name)).append("\n");
    sb.append("    version: ").append(toIndentedString(version)).append("\n");
    sb.append("    configurationFileName: ").append(toIndentedString(configurationFileName)).append("\n");
    sb.append("    description: ").append(toIndentedString(description)).append("\n");
    sb.append("    id: ").append(toIndentedString(id)).append("\n");
    sb.append("    imageTag: ").append(toIndentedString(imageTag)).append("\n");
    sb.append("}");
    return sb.toString();
  }

  /**
   * Convert the given object to string with each line indented by 4 spaces
   * (except the first line).
   */
  private String toIndentedString(java.lang.Object o) {
    if (o == null) {
      return "null";
    }
    return o.toString().replace("\n", "\n    ");
  }

}

File: SampleCode/RestApi/Net.RestSharp/EmbyClient.Dotnet/Model/PluginsPluginInfo.cs
/*
 * EmbyClient.Dotnet
 */

using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Text.RegularExpressions;
using System.Collections;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Runtime.Serialization;
using Newtonsoft.Json;
using Newtonsoft.Json.Converters;
using SwaggerDateConverter = EmbyClient.Dotnet.Client.SwaggerDateConverter;

namespace EmbyClient.Dotnet.Model
{
    /// <summary>
    /// This is a serializable stub class that is used by the api to provide information about installed plugins.  
    /// </summary>
    [DataContract]
        public partial class PluginsPluginInfo :  IEquatable<PluginsPluginInfo>
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="PluginsPluginInfo" /> class.
        /// </summary>
        /// <param name="name">The name..</param>
        /// <param name="version">The version..</param>
        /// <param name="configurationFileName">The name of the configuration file..</param>
        /// <param name="description">The description..</param>
        /// <param name="id">The unique id..</param>
        /// <param name="imageTag">The image URL..</param>
        public PluginsPluginInfo(string name = default(string), string version = default(string), string configurationFileName = default(string), string description = default(string), string id = default(string), string imageTag = default(string))
        {
            this.Name = name;
            this.Version = version;
            this.ConfigurationFileName = configurationFileName;
            this.Description = description;
            this.Id = id;
            this.ImageTag = imageTag;
        }
        
        /// <summary>
        /// The name.
        /// </summary>
        /// <value>The name.</value>
        [DataMember(Name="Name", EmitDefaultValue=false)]
        public string Name { get; set; }

        /// <summary>
        /// The version.
        /// </summary>
        /// <value>The version.</value>
        [DataMember(Name="Version", EmitDefaultValue=false)]
        public string Version { get; set; }

        /// <summary>
        /// The name of the configuration file.
        /// </summary>
        /// <value>The name of the configuration file.</value>
        [DataMember(Name="ConfigurationFileName", EmitDefaultValue=false)]
        public string ConfigurationFileName { get; set; }

        /// <summary>
        /// The description.
        /// </summary>
        /// <value>The description.</value>
        [DataMember(Name="Description", EmitDefaultValue=false)]
        public string Description { get; set; }

        /// <summary>
        /// The unique id.
        /// </summary>
        /// <value>The unique id.</value>
        [DataMember(Name="Id", EmitDefaultValue=false)]
        public string Id { get; set; }

        /// <summary>
        /// The image URL.
        /// </summary>
        /// <value>The image URL.</value>
        [DataMember(Name="ImageTag", EmitDefaultValue=false)]
        public string ImageTag { get; set; }

        /// <summary>
        /// Returns the string presentation of the object
        /// </summary>
        /// <returns>String presentation of the object</returns>
        public override string ToString()
        {
            var sb = new StringBuilder();
            sb.Append("class PluginsPluginInfo {\n");
            sb.Append("  Name: ").Append(Name).Append("\n");
            sb.Append("  Version: ").Append(Version).Append("\n");
            sb.Append("  ConfigurationFileName: ").Append(ConfigurationFileName).Append("\n");
            sb.Append("  Description: ").Append(Description).Append("\n");
            sb.Append("  Id: ").Append(Id).Append("\n");
            sb.Append("  ImageTag: ").Append(ImageTag).Append("\n");
            sb.Append("}\n");
            return sb.ToString();
        }
  
        /// <summary>
        /// Returns the JSON string presentation of the object
        /// </summary>
        /// <returns>JSON string presentation of the object</returns>
        public virtual string ToJson()
        {
            return JsonConvert.SerializeObject(this, Formatting.Indented);
        }

        /// <summary>
        /// Returns true if objects are equal
        /// </summary>
        /// <param name="input">Object to be compared</param>
        /// <returns>Boolean</returns>
        public override bool Equals(object input)
        {
            return this.Equals(input as PluginsPluginInfo);
        }

        /// <summary>
        /// Returns true if PluginsPluginInfo instances are equal
        /// </summary>
        /// <param name="input">Instance of PluginsPluginInfo to be compared</param>
        /// <returns>Boolean</returns>
        public bool Equals(PluginsPluginInfo input)
        {
            if (input == null)
                return false;

            return 
                (
                    this.Name == input.Name ||
                    (this.Name != null &&
                    this.Name.Equals(input.Name))
                ) && 
                (
                    this.Version == input.Version ||
                    (this.Version != null &&
                    this.Version.Equals(input.Version))
                ) && 
                (
                    this.ConfigurationFileName == input.ConfigurationFileName ||
                    (this.ConfigurationFileName != null &&
                    this.ConfigurationFileName.Equals(input.ConfigurationFileName))
                ) && 
                (
                    this.Description == input.Description ||
                    (this.Description != null &&
                    this.Description.Equals(input.Description))
                ) && 
                (
                    this.Id == input.Id ||
                    (this.Id != null &&
                    this.Id.Equals(input.Id))
                ) && 
                (
                    this.ImageTag == input.ImageTag ||
                    (this.ImageTag != null &&
                    this.ImageTag.Equals(input.ImageTag))
                );
        }

        /// <summary>
        /// Gets the hash code
        /// </summary>
        /// <returns>Hash code</returns>
        public override int GetHashCode()
        {
            unchecked // Overflow is fine, just wrap
            {
                int hashCode = 41;
                if (this.Name != null)
                    hashCode = hashCode * 59 + this.Name.GetHashCode();
                if (this.Version != null)
                    hashCode = hashCode * 59 + this.Version.GetHashCode();
                if (this.ConfigurationFileName != null)
                    hashCode = hashCode * 59 + this.ConfigurationFileName.GetHashCode();
                if (this.Description != null)
                    hashCode = hashCode * 59 + this.Description.GetHashCode();
                if (this.Id != null)
                    hashCode = hashCode * 59 + this.Id.GetHashCode();
                if (this.ImageTag != null)
                    hashCode = hashCode * 59 + this.ImageTag.GetHashCode();
                return hashCode;
            }
        }

    }
}
